### PR TITLE
Enable all services with labels to open service ui, ref. DCOS-8533

### PR DESCRIPTION
**Pull request by mlunoe**
These are the three affected places in the UI:

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @jfurrow

File: src/js/structs/__tests__/Service-test.js [hunk 1]
```diff
@@ -786,4 +786,65 @@ describe('Service', function () {
...
+          // DCOS_SERVICE_PORT_INDEX: '80',
+          DCOS_SERVICE_SCHEME: 'https'
+        }
+      });
+      let service3 = new Service({
+        foo: 'bar',
+        labels: {
+          DCOS_SERVICE_NAME: 'baz',
+          // DCOS_SERVICE_PORT_INDEX: '80',
+          // DCOS_SERVICE_SCHEME: 'https'
```
Comment (jfurrow) [inline src/js/structs/__tests__/Service-test.js]: Did you intend to keep these commented out?

**Comment by mlunoe** [comment]
**mesosphere-ci** Retest this please

File: src/js/structs/__tests__/Service-test.js (hunk 1 above)
Comment (mlunoe) [inline src/js/structs/__tests__/Service-test.js]: I actually did, because it made it more explanatory to me what was going on

**Comment by jfurrow** [comment]
I'm receiving the following error:

**Comment by mlunoe** [comment]
Wow, how did I miss that one. Should be fixed now

**Comment by jfurrow** [comment]
@mlunoe I'm still receiving the same error when trying to load the `ServicesTable`

**Comment by jfurrow** [comment]
Nice work @mlunoe, works as expected!

File: src/js/components/ServiceList.js [hunk 2]
```diff
@@ -93,7 +92,7 @@ let ServiceList = React.createClass({
             content: (
               <a key="title"
                 onClick={this.handleServiceClick.bind(this, service)}
-                href={Cluster.getServiceLink(service.getName())}
+                href={service.getWebURL()}
```
Comment (philipnrmn) [inline src/js/components/ServiceList.js]: let's cache this instead of calling it twice

File: src/js/components/ServiceInfo.js [hunk 3]
```diff
@@ -71,11 +69,11 @@ class ServiceInfo extends React.Component {
         transitionName="dropdown-menu" />
     ];
 
-    if (service instanceof Framework && service.getWebURL()) {
+    if (service.getWebURL()) {
       actionButtons.unshift(
         <a className="button button-primary flush-bottom"
           key="service-link"
-          href={Cluster.getServiceLink(service.getName())} target="_blank"
+          href={service.getWebURL()} target="_blank"
```
Comment (philipnrmn) [inline src/js/components/ServiceInfo.js]: let's cache this instead of calling it twice

File: src/js/components/ServicesTable.js [hunk 4]
```diff
@@ -147,13 +146,14 @@ var ServicesTable = React.createClass({
...
   getOpenInNewWindowLink(service) {
-    if (!(service instanceof Framework) || !service.getWebURL()) {
+    if (!(service instanceof Service) || !service.getWebURL()) {
       return null;
     }
 
     return (
       <a className="table-display-on-row-hover"
-        href={Cluster.getServiceLink(service.getName())} target="_blank"
+        href={service.getWebURL()}
```
Comment (philipnrmn) [inline src/js/components/ServicesTable.js]: let's cache this instead of calling it twice

File: src/js/components/ServiceInfo.js (hunk 3 above)
Comment (mlunoe) [inline src/js/components/ServiceInfo.js]: done

File: src/js/components/ServiceList.js (hunk 2 above)
Comment (mlunoe) [inline src/js/components/ServiceList.js]: This is called in two different functions

File: src/js/components/ServicesTable.js (hunk 4 above)
Comment (mlunoe) [inline src/js/components/ServicesTable.js]: Added a comment about why this is not possible, and why we need the check for `Service`

File: src/js/components/ServiceList.js (hunk 2 above)
Comment (philipnrmn) [inline src/js/components/ServiceList.js]: OK let's create a ServiceWebURLStore to hold the denormalized data then

**Pull request merged**